src/daemons/MetaDaemonInit.cpp

```diff
@@ -145,13 +145,8 @@ std::unique_ptr<nebula::kvstore::KVStore> initKV(std::vector<nebula::HostAddr> p
     LOG(ERROR) << "Meta version is invalid";
     return nullptr;
   } else if (version == nebula::meta::MetaVersion::V1) {
-    auto ret = nebula::meta::MetaVersionMan::updateMetaV1ToV2(engine);
-    if (!ret.ok()) {
-      LOG(ERROR) << "Update meta from V1 to V2 failed " << ret;
-      return nullptr;
-    }
-
-    nebula::meta::MetaVersionMan::setMetaVersionToKV(engine, nebula::meta::MetaVersion::V2);
+    LOG(ERROR) << "Can't upgrade meta from V1 to V3";
+    return nullptr;
   } else if (version == nebula::meta::MetaVersion::V2) {
```

Comment:
Why still exists `V2`, only V1 or V3 is return from `getMetaVersionFromKV`. So where do we call `updateMetaV2ToV3`?

Reply:
@HarrisChu , in previous test, does meta upgrade successfully? Something must be wrong here, either code or test

Reply:
if  kMetaVersionKey is existed, version will fetch from kv store

Reply:
Never mind, my bad